### PR TITLE
Language.py: Ignore whitespace in lookup

### PR DESCRIPTION
**Pull request by supergr35**
### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference).
- [x] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### Reviewers

Added a new strip() method to
the parse_lang_str function's
try statement to eliminate whitespace

Closes https://github.com/coala/coala/issues/3068

### CODE REVIEW

**Comment by ashwini0529** [comment]
@supergr35 You can run `coala` to get rid of such errors.

**Comment by Mixih** [comment]
All reviews on hold because of circle acting up. Please be patient as they work to fix their problems :) Also, marking WIP because errors.

File: coalib/bearlib/languages/Language.py [hunk 1]
```diff
@@ -39,6 +39,10 @@ def parse_lang_str(string):
     versions = list(map(convert_int_float, str_versions))
     try:
         name, version = name.rsplit(maxsplit=1)
+        if "Objective C" in name:
```
Comment (Mixih) [inline coalib/bearlib/languages/Language.py]: I really don't think this is what sils intended @sils you are summoned.

File: coalib/bearlib/languages/Language.py [hunk 2]
```diff
@@ -39,6 +39,7 @@ def parse_lang_str(string):
     versions = list(map(convert_int_float, str_versions))
     try:
         name, version = name.rsplit(maxsplit=1)
+        name = "".join(name.split())
```
Comment (Mixih) [inline coalib/bearlib/languages/Language.py]: Hmm no that doesn't do it either.. You must remove white-space from the internal language string being compared... Amending the doc-string doesn't fix.

**Comment by sils** [comment]
@supergr35 did you see @Mixih s last comment?

**Comment by supergr35** [comment]
@sils Yes, I have, but I might need some help with implementation

File: coalib/bearlib/languages/Language.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 import re
 from operator import itemgetter
 
-from coalib.settings.Annotations import typechain
+from coalib.misc.Annotations import typechain
```
Comment (Mixih) [inline coalib/bearlib/languages/Language.py]: this is just weird.....

File: coalib/bearlib/languages/Language.py [hunk 4]
```diff
@@ -149,7 +147,7 @@ class Sub(cls, metaclass=SubLanguageMeta):
                 aliases = tuple(sorted(getattr(arg, 'aliases', ())))
                 _attributes = {name: member for name, member in getmembers(arg)
                                if not name.startswith('_')
-                               and name not in forbidden_attributes}
+                               and not name in forbidden_attributes}
```
Comment (Mixih) [inline coalib/bearlib/languages/Language.py]: was this an intentional change?

File: coalib/bearlib/languages/Language.py [hunk 5]
```diff
@@ -342,16 +340,7 @@ def __getattr__(self, item):
         if len(self.versions) > 1:
             raise AttributeError('You have to specify ONE version of your '
                                  'language to retrieve attributes for it.')
-        try:
```
Comment (Mixih) [inline coalib/bearlib/languages/Language.py]: why this deletion?

**Comment by arush0311** [comment]
Use imperative tense in your commit message.
Perhaps this commit message is more helpful:
```
This commit strips the whitespace(s) after the language
name after extracting versions from language string.
```

**Comment by nemaniarjun** [comment]
Marking WIP, changes to be made in commit message. 

The [commit guideline](http://coala.io/commit) is usually helpful :)

File: coalib/bearlib/languages/Language.py [hunk 6]
```diff
@@ -39,6 +39,7 @@ def parse_lang_str(string):
     versions = list(map(convert_int_float, str_versions))
     try:
         name, version = name.rsplit(maxsplit=1)
+        name.strip()
```
Comment (Mixih) [inline coalib/bearlib/languages/Language.py]: this isn't doing anything as the result is not being assigned.

**Comment by Mixih** [comment]
ack 3b08c45

**Comment by Mixih** [comment]
ack 3b08c458d11cf9ceb0c43a2c9b2207a684e8f780

**Comment by Mixih** [comment]
Pending @sils review as he requested to review

**Comment by Mixih** [comment]
acked by sils  https://gitter.im/coala/coala?at=5881275b300f220a660b85f5

**Comment by Mixih** [comment]
@rultor merge

**Comment by Mixih** [comment]
please rebase

**Comment by Mixih** [comment]
reack 48192a1

**Comment by Mixih** [comment]
@rultor merge

**Pull request merged**